math/genvector/inc/Math/GenVector/DisplacementVector3D.h

```diff
@@ -333,13 +335,26 @@ namespace ROOT {
       Scalar Perp2() const { return fCoordinates.Perp2();}
 
       /**
-         return unit vector parallel to this
+         return unit vector parallel to this (scalar)
       */
-      DisplacementVector3D  Unit() const {
-        Scalar tot = R();
+      template< typename SCALAR = Scalar >
```

Comment:
So if I understand correctly the choice is between:
```
template< typename SCALAR = Scalar >
typename std::enable_if< std::is_arithmetic<SCALAR>::value, DisplacementVector3D >::type
Unit() const {
```
and
```
template <typename SCALAR = Scalar, typename = std::enable_if< std::is_arithmetic<SCALAR>::value >
DisplacementVector3 
Unit() const {
```
If that is correct, I prefer the 2nd one as the first one give the wrong impression that the return type is dependent on the template parameter (unless you actually read the template detail :) ) while the first clearly indicate that there is a limitation on the SCALAR type/template-parameter.